tests/Endpoints/TenantTokenTest.php

```diff
@@ -0,0 +1,171 @@
+        $tokenClient = new Client($this->host, $token);
+        $response = $tokenClient->index('tenantToken')->search('');
+
+        $this->assertArrayHasKey('hits', $response->toArray());
+        $this->assertArrayHasKey('query', $response->toArray());
+        $this->expectException(ApiException::class);
+        $response = $tokenClient->index('tenantTokenDuplicate')->search('');
+    }
+
+    public function testGenerateTenantTokenWithApiKey(): void
+    {
+        $options = [
+            'apiKey' => $this->privateKey,
+        ];
+        $token = $this->client->generateTenantToken(['*'], $options);
+        $tokenClient = new Client($this->host, $token);
+        $response = $tokenClient->index('tenantToken')->search('');
+
+        $this->assertArrayHasKey('hits', $response->toArray());
+        $this->assertArrayHasKey('offset', $response->toArray());
+        $this->assertArrayHasKey('limit', $response->toArray());
+        $this->assertArrayHasKey('processingTimeMs', $response->toArray());
+        $this->assertArrayHasKey('query', $response->toArray());
+    }
+
+    public function testGenerateTenantTokenWithExpiresAt(): void
+    {
+        /* @phpstan-ignore-next-line */
+        $date = new DateTime();
+        $tomorrow = $date->modify('+1 day');
+        $options = [
+            'expiresAt' => $tomorrow,
+        ];
+        $token = $this->privateClient->generateTenantToken(['*'], $options);
+        $tokenClient = new Client($this->host, $token);
+        $response = $tokenClient->index('tenantToken')->search('');
+
+        $this->assertArrayHasKey('hits', $response->toArray());
+        $this->assertArrayHasKey('offset', $response->toArray());
+        $this->assertArrayHasKey('limit', $response->toArray());
+        $this->assertArrayHasKey('processingTimeMs', $response->toArray());
+        $this->assertArrayHasKey('query', $response->toArray());
+    }
+
+    public function testGenerateTenantTokenWithSearchRulesEmptyArray(): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $token = $this->privateClient->generateTenantToken([]);
+    }
+
+    public function testGenerateTenantTokenWithBadExpiresAt(): void
+    {
+        /* @phpstan-ignore-next-line */
+        $date = new DateTime();
+        $yesterday = $date->modify('-1 day');
+        $options = [
+            'expiresAt' => $yesterday,
+        ];
+
+        $this->expectException(InvalidArgumentException::class);
+        $token = $this->privateClient->generateTenantToken(['*'], $options);
+    }
+
+    public function testGenerateTenantTokenWithNoApiKey(): void
+    {
+        $client = new Client($this->host);
+        $this->expectException(InvalidArgumentException::class);
+        $token = $client->generateTenantToken(['*']);
+    }
+
+    public function testGenerateTenantTokenWithEmptyApiKey(): void
+    {
+        $client = new Client($this->host);
```

Comment:
Could you please give a look here:?

- http://xunitpatterns.com/Four%20Phase%20Test.html
- http://blog.plataformatec.com.br/2014/04/improve-your-test-readability-using-the-xunit-structure/

This is a way to improve the readability of the tests we could apply everywhere :)

Reply:
For this case you will have this (and for the `testGenerateTenantTokenWithNoApiKey` as well):

```php
$this->expectException(InvalidArgumentException::class);

$client = new Client($this->host);
$client->generateTenantToken(['*'], ['apiKey' => '']);
```